RUFAS/simulation_engine.py

```diff
@@ -123,27 +124,78 @@ def _run_simulation_main_loop(self) -> None:
 
     def _daily_simulation(self) -> None:
         """Executes the daily simulation routines."""
-
-        # TODO: remove this code after Animal and Feed Storage modules are connected - #1878
-        if self.is_end_to_end_test_run:
-            process_degradations_today = self.time.current_julian_day % FEED_DEGRADATION_INTERVAL_LENGTH == 0
-            if process_degradations_today:
-                self.feed_manager.process_degradations(self.weather, self.time)
-
-        self.animal_manager.daily_updates(self.feed, self.weather, self.time)
-        all_pen_manure_data = self.animal_manager.collect_pen_manure_data()
-        self.manure_manager.daily_update(all_pen_manure_data, self.animal_manager.simulation_day)
         manure_applications = self.generate_daily_manure_applications()
         harvested_crops = self.field_manager.daily_update_routine(self.weather, self.time, manure_applications)
+        next_harvest_dates: dict[str, date | None] = {}
         for harvested_crop in harvested_crops:
             self.feed_manager.receive_crop(harvested_crop.harvested_crop, harvested_crop.storage_type)
-        routines.daily_feed_routine(self.feed, self.field_manager, self.animal_manager)
+            if harvested_crop.harvested_crop.config_name not in next_harvest_dates.keys():
```

Comment:
```suggestion
            if harvested_crop.harvested_crop.config_name not in next_harvest_dates:
```
Avoids creating unnecessary dict keys view object.